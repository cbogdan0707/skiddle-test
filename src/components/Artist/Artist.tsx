import React from 'react';
import { config } from '../../utils/config';
import { useGetEventsOrArtists } from '../../utils/hooks';

export const Artist = ({ match }: { match: { params: { id: string } } }) => {
  const { hasError, response } = useGetEventsOrArtists(
    config.eventArtistsUrl,
    match,
    config.apiKey
  );

  return (
    <div>
      {hasError && <div data-testid="fetch-error">Error: {hasError}</div>}
      {response.results && (
        <>
          <div className="skiddle-u-text-align-center skiddle-u-background-primary">
            <img
              className="skiddle-c-artist-image skiddle-u-mt-large"
              alt="artist logo"
              src={response.results.imageurl}
            />
            <h1 className="skiddle-u-color-white skiddle-u-mv-large">
              {response.results.name}
            </h1>
            <p className="skiddle-c-artist-description">
              {response.results.description}
            </p>
          </div>
        </>
      )}
    </div>
  );
};
