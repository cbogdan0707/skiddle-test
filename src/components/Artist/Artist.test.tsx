import React from 'react';
import { render } from '@testing-library/react';
import { Artist } from './Artist';

describe('Artist details', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Artist match={{ params: { id: 2333333 } }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
