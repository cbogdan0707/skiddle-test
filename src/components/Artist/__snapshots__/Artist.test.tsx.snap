// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist details should match snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
